### ISSUE TITLE
Parallel runtime tests execution

### ISSUE THREAD

**Issue by KvanTTT** [antlr/antlr4#3370]
I don't understand why parallel tests execution still is not enabled. It should be quite easy to implement since each test uses a separated directory and operates in an isolated process. Parallel tests can greatly improve performance because they may utilize all processor cores.

**Comment by ericvergnaud** [comment]
Free CIs usually only provide single core containers so enabling parallel testing ends up being slower. This was definitely the case with Travis CI. Didn't try it on our latest CIs though.

**Comment by KvanTTT** [comment]
Also, it is usefull on local machines without limitation on number of cores.

**Comment by KvanTTT** [comment]
BTW, now I'm still not able to set up tests, I'm not a very experienced with maven.

**Comment by KvanTTT** [comment]
It seems like at least CircleCI has 2 CPUs: https://app.circleci.com/pipelines/github/antlr/antlr4/678/workflows/005d374e-8b5b-40ae-a720-c04b03a6928b/jobs/7728/resources

**Comment by KvanTTT** [comment]
AppVeyor also has 2 CPU cores: https://help.appveyor.com/discussions/questions/11179-how-many-cores-and-threads-can-be-used-in-free-appveyor-build

**Comment by ericvergnaud** [comment]
you can try amending the test scripts, there’s a maven cmd line option I believe

**Comment by KvanTTT** [comment]
Yes, I'll try to fix them after fixing dotnet and go pipelines on Linux. Also, it seems like parallel execution is already turned on for CircleCI.
